app/src/main/java/com/android/sample/ui/activity/CreateActivity.kt

```diff
@@ -380,61 +460,98 @@ fun CreateActivityScreen(
+        override fun onCaptureSuccess(image: ImageProxy) {
+          super.onCaptureSuccess(image)
+
+          val matrix = Matrix().apply { postRotate(image.imageInfo.rotationDegrees.toFloat()) }
+          val rotatedBitmap =
+              Bitmap.createBitmap(image.toBitmap(), 0, 0, image.width, image.height, matrix, true)
 
-// @Composable
-// fun Carousel() {
-//    Row(
-//        modifier = Modifier
-//            .fillMaxWidth()
-//            .height(135.dp)
-//            .padding(8.dp),
-//        verticalAlignment = Alignment.CenterVertically
-//    ) {
-//        LazyRow(
-//            modifier = Modifier
-//                .width(340.dp)
-//                .height(135.dp),
-//        ) {
-//            items(items.size) { index ->
-//                AsyncImage(
-//                    model = items[index], // Utilise l'URL de l'image
-//                    contentDescription = "image $index",
-//                    contentScale = ContentScale.Crop,
-//                    modifier = Modifier.padding(8.dp)
-//                )
-//            }
-//        }
-//        Spacer(modifier = Modifier.width(16.dp))
-//        Column(
-//            modifier =
-//            Modifier
-//                .padding(16.dp)
-//                .size(30.dp)
-//                .background(Color(0xFFFFFFFF)), // Use size modifier for simplicity
-//            verticalArrangement = Arrangement.Center,
-//            horizontalAlignment = Alignment.End // Center the icon horizontally
-//        ) {
-//            FloatingActionButton(
-//                content = {
-//                    Icon(
-//                        imageVector = Icons.Outlined.AddCircle,
-//                        contentDescription = "Add a new image"
-//                    )
-//                },
-//                containerColor = Color(0xFFFFFFFF),
-//                onClick = { /*TODO*/ },
-//                modifier = Modifier
-//                    .size(50.dp)
-//                    .background(Color(0xFFFFFFFF)),
-//            )
-//
-//            Text(
-//                text = "Add Image",
-//                modifier = Modifier.padding(8.dp),
-//                color = Color(0xFF000000),
-//                style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 12.sp),
-//            )
-//        }
-//    }
-// }
+          onPhotoTaken(rotatedBitmap)
+        }
+
+        override fun onError(exception: ImageCaptureException) {
+          super.onError(exception)
+          Toast.makeText(
+                  applicationContext,
+                  "Error taking picture: ${exception.message}",
+                  Toast.LENGTH_SHORT)
+              .show()
+        }
+      })
+}
+
+fun bitmapToBase64(bitmap: Bitmap): String {
+  val byteArrayOutputStream = ByteArrayOutputStream()
+  bitmap.compress(Bitmap.CompressFormat.PNG, 70, byteArrayOutputStream)
+  val byteArray = byteArrayOutputStream.toByteArray()
+  return Base64.encodeToString(byteArray, Base64.DEFAULT)
+}
```

Comment:
Function that will be used to convert the images from bitmap to strings to be able to store in the firestore.